core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -77,8 +77,8 @@ public Query rewrite(Query original) throws IOException {
     public Weight createNormalizedWeight(Query query, boolean needsScores) throws IOException {
         try {
             // if scores are needed and we have dfs data then use it
-            if (dfSource != null && needsScores) {
-                return dfSource.createNormalizedWeight(query, needsScores);
+            if (aggregatedDfs != null && needsScores) {
+                return super.createNormalizedWeight(query, needsScores);
             }
             return in.createNormalizedWeight(query, needsScores);
```

Comment:
Wouldn't the 3 above lines be equivalent to just doing: `return super.createNormalizedWeight(query, needsScores);`?